thrust/examples/run_length_decoding_single_pass.cmake

```diff
@@ -0,0 +1,41 @@
+# This example uses a single-pass algorithm that performs side effects during scan.
+# The algorithm relies on the scan functor being called exactly once per element with the final
+# scan value, which is guaranteed by CUDA but not by CPU-parallel backends like OMP/TBB where
+# the functor may be called multiple times with intermediate values during parallel reduction.
```

Comment:
Suggestion: I think we should also put this disclaimer into the example source code.

Reply:
That comment may not be entirely accurate, it's a question of how many times the output iterator is written to, because that's what invokes the functor.